sdk/python/kfp/cli/install_cli/kfp_installer.py

```diff
@@ -0,0 +1,191 @@
+  folder = tempfile.mkdtemp()
+  cluster_scoped_folder = os.path.join(folder, 'cluster-scoped')
+  os.mkdir(cluster_scoped_folder)
+
+  print("Created temp directory: {0}".format(folder))
+  return folder, cluster_scoped_folder
+
+def cleanup_kustomize_folder(folder):
+  shutil.rmtree(folder)
+  print("Deleted temp directory: {0}".format(folder))
+
+def install_cluster_scoped(cluster_scoped_folder, namespace, install_version):
+  print('Preparing cluster-scoped workspace')
+  kustomize_file = os.path.join(cluster_scoped_folder, 'kustomization.yaml')
+  with open(kustomize_file, 'w') as file:
+    utils.write_line(file, 'apiVersion: kustomize.config.k8s.io/v1beta1')
+    utils.write_line(file, 'kind: Kustomization')
+    utils.write_line(file, 'bases:')
+    utils.write_line(file, '  - github.com/kubeflow/pipelines/manifests/kustomize/cluster-scoped-resources{0}'.format(
+        version_ref(install_version)))
+    utils.write_line(file, '')
+    utils.write_line(file, 'configMapGenerator:')
+    utils.write_line(file, '  - name: pipeline-cluster-scoped-install-config')
+    utils.write_line(file, '    env: params.env')
+    utils.write_line(file, '    behavior: merge')
+
+  param_file = os.path.join(cluster_scoped_folder, 'params.env')
+  with open(param_file, 'w') as file:
+    utils.write_line(file, 'namespace={0}'.format(namespace))
+
+  cmd = 'kubectl apply -k {0}'.format(cluster_scoped_folder)
+  print("Executing command to apply cluster scoped resources: {0}".format(cmd))
+  result = executer.execute_subprocess(cmd)
+  if result.returncode != 0:
+    utils.print_error("Can't apply cluster scoped resources.")
+    print("Please check your inputs and report issue: {0}".format(
+        "https://github.com/kubeflow/pipelines/issues/new?template=BUG_REPORT.md"))
+    exit(1)
+  else:
+    utils.print_success("Successfully applied cluster scoped resources.")
+
+def wait_for_cluster_scoped_ready():
+  cmd = 'kubectl wait crd/applications.app.k8s.io --for condition=established --timeout=60s'
+  print("Executing command to wait for cluster scoped resources ready: {0}".format(cmd))
+  result = executer.execute_subprocess(cmd)
+  if result.returncode != 0:
+    print('Failed or timeout with code: {0}. Retry once'.format(result.returncode))
+    result = executer.execute_subprocess(cmd)
+    if result.returncode != 0:
+      utils.print_error("Wait for cluster scoped resources be ready but failed.")
+      exit(1)
+  utils.print_success("Cluster scoped resources ready.")
+
+def install_namespace_scoped(folder, gcp_project_id, gcs_default_bucket, instance_name, namespace,
+    enable_managed_storage, cloud_sql_instance_name, cloud_sql_username, cloud_sql_password,
+    install_version):
+  print('Preparing namespace-scoped workspace')
+  kustomize_file = os.path.join(folder, 'kustomization.yaml')
+  with open(kustomize_file, 'w') as file:
+    utils.write_line(file, 'apiVersion: kustomize.config.k8s.io/v1beta1')
```

Comment:
You can just use python's miltiline strings here. You can also `.format()` them.

Reply:
prefer use current format to well isolate the lines (for easy format in case we have more customization).

Another option is to use template. However I didn't find a good way to well replace values (same worry for the multiline string)

Reply:
left a TODO for it, agree current single line is not perfact